jadx-core/src/main/java/jadx/core/dex/visitors/SimplifyVisitor.java

```diff
@@ -201,7 +204,15 @@ private static InsnNode convertInvoke(MethodNode mth, InsnNode insn) {
 						}
 
 						for (; argInd < len; argInd++) {  // Add the .append(xxx) arg string to concat
-							concatInsn.addArg(chain.get(argInd).getArg(1));
+							InsnNode node = chain.get(argInd);
+							MethodInfo method = ((CallMthInterface) node).getCallMth();
```

Comment:
Add `instanceof` check before the cast, it will make code much safer.
And next time instead of merging just rebase your commits on latest master, also fix for the previous commit is better to squash together, so you will get 2 nice commits on top of master :)

Reply:
The missing `instanceof` check is on purpose. If we encounter such a case most likely can't simplify the chain. It will then raise an Exception so that anybody may notice it and we can take a closer look onto it. Therefore the effect is the same the only difference is that without the instanceof check nothing is logged.

Rebase: According what I have read you should not do this once you have pushed your commits to a public repo. And as the branch this PR bases on is public... 

Therefore I will keep on using git as I do - at least most of the times it does what I want. 
If they develop such a complex version control system they should make it more usable and intuitive to use.